Put SVG images into CSS

GuillaumeGomez opened the pull request:
Since CSS files are often cached, we can take advantage of this to reduce network usage.

Now how it works: we inline the SVG images into a CSS file we generate at compile-time to ensure this file only contains what we need to limit its use but also to generate equivalent Rust types so we know at compile-time that the SVG exists.

I checked on caniuse for `mask` here: https://caniuse.com/css-masks. It says that more than 89% of current web browsers supports it unprefixed (without `-webkit-`). With the prefix, it goes up to more than 96%. Just in case, I used both (thanks to HTML variables, we don't need to duplicate the SVG for both `mask`) so we should be on the safe side with more than 96% of our potential users being able to see the icons.

I compared the HTML size on https://docs.rs/sysinfo/0.30.12/sysinfo/index.html. On current docs.rs, the transferred HTML is 52KB, with this PR is goes down to 40KB.

GuillaumeGomez commented on src/web/csp.rs:
```diff
@@ -54,7 +54,7 @@ impl Csp {
         //
         // Images from other HTTPS origins are also temporary allowed until issue #66 is fixed.
-        result.push_str("; img-src 'self' https:");
+        result.push_str("; img-src 'self' https: data:");
```
This change allows to have inlined SVG in the CSS.

GuillaumeGomez replied:
If it's an issue, we can instead use URLs and have the SVG images stored in files.

GuillaumeGomez commented:
@syphar: Any idea why there are sqlx errors?

syphar commented on build.rs:
```diff
@@ -80,11 +84,194 @@ fn main() -> Result<()> {
+    let css_class = format!("f-a_{icon_name}_{icon_kind}");
+    css_output.push_str(&format!(
+        "\
+.{css_class} {{
```
what I wonder: 

isn't a CSS distribution with the icons something that font-awesome already provides? That we can just use?

syphar commented:
not finished yet with all review, but one bigger question

GuillaumeGomez commented on build.rs:
```diff
@@ -80,11 +84,194 @@ fn main() -> Result<()> {
     write_known_targets(out_dir)?;
     compile_syntax(out_dir).context("could not compile syntax files")?;
 
+    println!("cargo::rustc-check-cfg=cfg(icons_out_dir)");
```
It's used in `src/lib.rs` to import the generated Rust file.

syphar reviewed (changes requested):
Thank you for the work! 

I tried to test it locally, but I don't see any icons, probably because the fontawesome.css request leads to a 404

syphar commented:
Next thing from manual testing: 

new: 
<img width="337" alt="grafik" src="https://github.com/user-attachments/assets/88664ac1-370e-4c50-a9f6-5ef14a9ab3d7">

old: 
<img width="312" alt="grafik" src="https://github.com/user-attachments/assets/64d91988-6085-4fed-98f8-5435c000e769">


it seems like the clipboard icon is missing there? 

it's also missing on the crate details page: 
<img width="529" alt="grafik" src="https://github.com/user-attachments/assets/8d311b73-f77b-41c7-a934-557d97844376">


The rest of the tests look good

GuillaumeGomez commented:
Found the issues:
1. The CSS file where I added `svg-clipboard` was not included in all docs.rs pages. So I moved it into `font-awesome.css` instead because I already modified it (for the fonts paths) to make it work with docs.rs. And since it's an icon, made more sense to put it there.
2. The SVG image contained a jinja comment that I forgot to update to HTML comment.

We should really add more GUI tests. ^^'
